[Security Solution][Entity Analytics][Risk Score] Add Risk Scoring Alert Filtering - Frontend Implementation

abhishekbhatia1710 opened the pull request:
## Summary

This PR implements the frontend user interface for Risk Scoring Alert Filtering in Entity Analytics. Users can now create and manage KQL-based filters to target specific entity types (Users, Hosts, Services) during risk score calculations.

## Design Implementation

- Based on the [Figma mockups](https://www.figma.com/design/IKQr6HbQPAdFws9i7Ikcx9/-8.15-9.2--Entity-Analytics?node-id=7131-49715&t=8Z3JSBzT4pZ2HkXZ-0), this implementation provides:
- KQL Search Bar with field suggestions
- Filter Management with visual chips and entity targeting
- Save Functionality with persistent storage


## User Interface

### Filter Creation Flow

- User types in KQL search bar (e.g., user.name: "root")
- Field suggestions appear as they type
- Press Enter to create a filter chip
- Select target entities (Users, Hosts, Services)
- Click "Save changes" to persist filters

### Filter Management

- Remove Individual Filters: Cross button on each filter chip
- Remove Entire Filter: Cross button at the end of each filter row
- Visual Feedback: Green values, proper spacing, and clear hierarchy

Screenshot : 

<img width="1913" height="881" alt="image" src="https://github.com/user-attachments/assets/69c375d5-e3de-41ae-bd36-2dd6ca24fe9e" />


<img width="1918" height="930" alt="image" src="https://github.com/user-attachments/assets/b5c6cf31-e8bc-404d-aec8-6c929d0ce2f4" />


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)
- [ ] Review the [backport guidelines](https://docs.google.com/document/d/1VyN5k91e5OVumlc0Gb9RPa3h1ewuPE705nRtioPiTvY/edit?usp=sharing) and apply applicable `backport:*` labels.

abhishekbhatia1710 committed:
feat: Add Risk Scoring Alert Filtering - Backend Implementation

- Add filters field to risk-engine-configuration saved object
- Implement migration logic for existing configurations
- Enhance risk scoring logic with entity-specific KQL filters
- Update API endpoints to support filter configuration and preview
- Add comprehensive unit tests for new functionality
- Maintain backward compatibility with existing configurations

Closes: elastic/security-team#13606

abhishekbhatia1710 committed:
Adding filters to the esql risk scoring

abhishekbhatia1710 committed:
Adding filters for the preview API

abhishekbhatia1710 committed:
Removing repeated code

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/check_mappings_update --fix'

kibanamachine committed:
[CI] Auto-commit changed files from 'yarn openapi:bundle'

kibanamachine committed:
[CI] Auto-commit changed files from 'make api-docs'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/jest_integration -u src/core/server/integration_tests/ci_checks'

abhishekbhatia1710 committed:
Implement entity-specific filtering for risk scoring

- Refactor both ES|QL and scripted metrics implementations to create separate queries for each entity type
- Apply custom filters only to specified entity types using entity_types field
- Fix TypeScript errors with proper type casting using Record<string, unknown> instead of any
- Ensure filters are applied as inclusion filters (must match) not exclusion filters

abhishekbhatia1710 committed:
Changing the filter query to filter out instead of filter in

abhishekbhatia1710 committed:
Fixing types

abhishekbhatia1710 committed:
Fixing tests :  updating mappings version

abhishekbhatia1710 committed:
feat: Add KQL alert filtering UI for Entity Analytics Risk Scoring
- Add KQL search bar with field suggestions for alert filtering
- Implement filter chips with entity-specific targeting (Users/Hosts/Services)
- Add save functionality with localStorage persistence

abhishekbhatia1710 committed:
feat: Add KQL alert filtering UI for Entity Analytics Risk Scoring
- Add KQL search bar with field suggestions for alert filtering
- Implement filter chips with entity-specific targeting (Users/Hosts/Services)
- Add save functionality with localStorage persistence

elasticmachine commented:
Pinging @elastic/security-entity-analytics (Team:Entity Analytics)

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/348646)
* Commit: c987bc93ab953f66ba817d0d884e3c11c65e4631

### Failed CI Steps


### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/76f62f9ac82841f62672653b56c10f81a2278795) | [after](https://github.com/elastic/kibana/commit/c987bc93ab953f66ba817d0d884e3c11c65e4631) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8295 | 8297 | +2 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/76f62f9ac82841f62672653b56c10f81a2278795) | [after](https://github.com/elastic/kibana/commit/c987bc93ab953f66ba817d0d884e3c11c65e4631) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 11.1MB | 11.1MB | +6.1KB |



cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348646","number":348646,"commit":"c987bc93ab953f66ba817d0d884e3c11c65e4631"}],"number":348646}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Added changes to validations and KQL bar

abhishekbhatia1710 committed:
Added changes to validations and KQL bar

abhishekbhatia1710 committed:
Add filters to ReadRiskEngineSettingsResponse

- Settings route now returns filters array from saved object
- Updated schema to include filters in response
- Generated types updated to match schema

abhishekbhatia1710 committed:
Ensure filter exclusion logic in risk calculations

- Filters use must_not (exclusion) to filter out matching alerts
- Applied to both standard and ESQL calculation methods
- Aligns with customer use cases (filter out test data, service accounts, etc)

abhishekbhatia1710 committed:
Add frontend UI for alert filtering in risk scores

abhishekbhatia1710 committed:
Add frontend UI for alert filtering in risk scores

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/351431)
* Commit: 2cfdcbd369972568e7721a059e98e91a999cd7b2

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/351431#0199e8c9-e2dd-411c-be93-a0b5f60d5374)

### History
* :broken_heart: [Build #350580](https://buildkite.com/elastic/kibana-pull-request/builds/350580) failed 9014a177b2527dc62546b9675c1fc9224d0fe7eb
* :broken_heart: [Build #350176](https://buildkite.com/elastic/kibana-pull-request/builds/350176) failed 0243165936395b1324b268a9999f53a873f436f9
* :broken_heart: [Build #348646](https://buildkite.com/elastic/kibana-pull-request/builds/348646) failed c987bc93ab953f66ba817d0d884e3c11c65e4631

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351431","number":351431,"commit":"2cfdcbd369972568e7721a059e98e91a999cd7b2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350580","number":350580,"commit":"9014a177b2527dc62546b9675c1fc9224d0fe7eb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350176","number":350176,"commit":"0243165936395b1324b268a9999f53a873f436f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348646","number":348646,"commit":"c987bc93ab953f66ba817d0d884e3c11c65e4631"}],"number":351431}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Backend changes readded after resolving merge conflicts for the Risk score filtering

abhishekbhatia1710 committed:
Re-adding FE changes for Risk score filtering after resolving merge conflicts

abhishekbhatia1710 committed:
Re-adding FE changes for Risk score filtering after resolving merge conflicts

abhishekbhatia1710 committed:
Fixing Save bar visibility and filter persistence

abhishekbhatia1710 committed:
Fixing Save bar visibility and filter persistence

abhishekbhatia1710 committed:
TypeScript errors for filters field without backend types

abhishekbhatia1710 committed:
TypeScript errors for filters field without backend types

abhishekbhatia1710 committed:
Fixing types

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/351470)
* Commit: b40711d112a874f4b631b61c8ce475de54a85a87

### Failed CI Steps
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8b6-4015-ab13-5d72b2d30805)
* [Jest Tests #9](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e965-4859-4518-9192-0d9115d43a1e)
* [Jest Integration Tests #4](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8bb-40cf-8223-9b40d52a5dd9)
* [FTR Configs #19](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8d6-47bd-9fa0-9a7b729aa750)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8b6-4015-ab13-5d72b2d30805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8b6-4015-ab13-5d72b2d30805/artifacts/0199e965-2aed-49c6-afa8-41408a8435c7) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage calls the preview section with the toggle button selection as it changes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e965-4859-4518-9192-0d9115d43a1e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e965-4859-4518-9192-0d9115d43a1e/artifacts/0199e97f-7739-48e7-94fa-f8a18f07ef53) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage calls the preview section with the toggle button selection as it changes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8b6-4015-ab13-5d72b2d30805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8b6-4015-ab13-5d72b2d30805/artifacts/0199e965-2aed-4b39-91cc-82eb925e5c1f) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage has the major sections of the page visible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e965-4859-4518-9192-0d9115d43a1e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e965-4859-4518-9192-0d9115d43a1e/artifacts/0199e97f-7738-40b5-a4e3-5d3dd4d3753b) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage has the major sections of the page visible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8b6-4015-ab13-5d72b2d30805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8b6-4015-ab13-5d72b2d30805/artifacts/0199e965-2aed-4ef7-ae41-387c5f2742fa) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage toggles the save bar when making changes to the closed alerts toggle
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e965-4859-4518-9192-0d9115d43a1e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e965-4859-4518-9192-0d9115d43a1e/artifacts/0199e97f-7738-4751-a597-058789c56a30) Jest Tests #<span></span>9 / EntityAnalyticsManagementPage toggles the save bar when making changes to the closed alerts toggle
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8bb-40cf-8223-9b40d52a5dd9) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8bb-40cf-8223-9b40d52a5dd9/artifacts/0199e96d-0963-415f-ba1a-851c5e54fafb) Jest Integration Tests #<span></span>4 / Migration actions - serverless environment createIndex resolves left cluster_shard_limit_exceeded when the action would exceed the maximum normal open shards
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8b6-4015-ab13-5d72b2d30805) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8b6-4015-ab13-5d72b2d30805/artifacts/0199e965-2aec-4e1e-9991-41ce3b83261e) Jest Tests #<span></span>9 / search_bar should render search bar and fetch index pattern
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e965-4859-4518-9192-0d9115d43a1e) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e965-4859-4518-9192-0d9115d43a1e/artifacts/0199e97f-7739-4c69-8bb2-43d978c06f26) Jest Tests #<span></span>9 / search_bar should render search bar and fetch index pattern
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/351470#0199e94b-b8d6-47bd-9fa0-9a7b729aa750) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/351470/jobs/0199e94b-b8d6-47bd-9fa0-9a7b729aa750/artifacts/0199e96c-e807-4054-b53e-18fd2838119b) FTR Configs #<span></span>19 / Serverless Common UI - Dev Tools Search Profiler Editor With a test index profiles a simple query

### History
* :broken_heart: [Build #351456](https://buildkite.com/elastic/kibana-pull-request/builds/351456) failed 01ad9fdad8565ac7058db7e3ade5a1752b8956d4
* :broken_heart: [Build #351431](https://buildkite.com/elastic/kibana-pull-request/builds/351431) failed 2cfdcbd369972568e7721a059e98e91a999cd7b2
* :broken_heart: [Build #350580](https://buildkite.com/elastic/kibana-pull-request/builds/350580) failed 9014a177b2527dc62546b9675c1fc9224d0fe7eb
* :broken_heart: [Build #350176](https://buildkite.com/elastic/kibana-pull-request/builds/350176) failed 0243165936395b1324b268a9999f53a873f436f9
* :broken_heart: [Build #348646](https://buildkite.com/elastic/kibana-pull-request/builds/348646) failed c987bc93ab953f66ba817d0d884e3c11c65e4631

cc @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351470","number":351470,"commit":"b40711d112a874f4b631b61c8ce475de54a85a87"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351456","number":351456,"commit":"01ad9fdad8565ac7058db7e3ade5a1752b8956d4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351431","number":351431,"commit":"2cfdcbd369972568e7721a059e98e91a999cd7b2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350580","number":350580,"commit":"9014a177b2527dc62546b9675c1fc9224d0fe7eb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350176","number":350176,"commit":"0243165936395b1324b268a9999f53a873f436f9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/348646","number":348646,"commit":"c987bc93ab953f66ba817d0d884e3c11c65e4631"}],"number":351470}
buildkite-pr-comment-->

PhilippeOberti reviewed (approved):
Code review only, LGTM for the @elastic/security-threat-hunting-investigations team

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/api/hooks/use_preview_risk_scores.ts:
TODO: delete after merged

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/hooks/risk_score_configurable_risk_engine_settings_hooks.ts:
Can we reuse the types from the server?

TODO: delete after merged

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/hooks/risk_score_configurable_risk_engine_settings_hooks.ts:
This comment doesn't add any information
```suggestion
```

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/hooks/risk_score_configurable_risk_engine_settings_hooks.ts:
Please avoid type casting. Update `fetchRiskEngineSettings` to return the right type.
```suggestion
      const backendFilters = riskEngineSettings?.filters;
```

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/alert_filters_kql_bar.tsx:
unit tests?

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/alert_filters_kql_bar.tsx:
This could be a constant `['user', 'host', 'service']`

machadoum reviewed (changes requested):
Hey, Great effort!

Could you please add unit tests for this feature?

I would also advise keeping this PR open until the BE is merged, and then refactor it to reuse the BE types. Otherwise, we will have several types duplicated

I merged both PRs and ran some desk testing. This is what I found:

* When an error happens, there is an up arrow pointing to nowhere (----^).
<img width="804" height="262" alt="Image" src="https://github.com/user-attachments/assets/13d27fac-0b65-45b4-8067-aca05dc5980f" />

* The filter doesn't work for the preview endpoint

<img width="2144" height="1292" alt="Image" src="https://github.com/user-attachments/assets/0e5e62e7-d902-4c9e-b105-c86bd2af3ae6" />

* The HTML markdown doesn't support KQLs composed of several filters

<img width="1458" height="832" alt="Image" src="https://github.com/user-attachments/assets/f6dd260c-b760-4d6d-9dcb-a161432254e8" />

* When I remove all entity types, they magically show up again 😖 

https://github.com/user-attachments/assets/b999766f-65c2-448a-b91e-269ae71c14d1

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/353163)
* Commit: cb679380cd783a791099e0821f2df1cf32e31ac9

### Failed CI Steps
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0714-b0d9-4c48-b8b3-0ba09aa702b7)
* [Jest Tests #3](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0729-0f00-4853-97a2-bcf8b188724d)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0714-b0d9-4c48-b8b3-0ba09aa702b7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0714-b0d9-4c48-b8b3-0ba09aa702b7/artifacts/019a0728-f5ca-4337-9790-047cac22dcf9) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage calls the preview section with the toggle button selection as it changes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0729-0f00-4853-97a2-bcf8b188724d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0729-0f00-4853-97a2-bcf8b188724d/artifacts/019a073d-7fbe-498f-b240-19f7409c6f28) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage calls the preview section with the toggle button selection as it changes
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0714-b0d9-4c48-b8b3-0ba09aa702b7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0714-b0d9-4c48-b8b3-0ba09aa702b7/artifacts/019a0728-f5ca-4e33-976a-e764a5591bef) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage has the major sections of the page visible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0729-0f00-4853-97a2-bcf8b188724d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0729-0f00-4853-97a2-bcf8b188724d/artifacts/019a073d-7fbe-4040-ae81-52c551d904bf) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage has the major sections of the page visible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0714-b0d9-4c48-b8b3-0ba09aa702b7) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0714-b0d9-4c48-b8b3-0ba09aa702b7/artifacts/019a0728-f5c9-49b7-8846-199b123f4702) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage toggles the save bar when making changes to the closed alerts toggle
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/353163#019a0729-0f00-4853-97a2-bcf8b188724d) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/353163/jobs/019a0729-0f00-4853-97a2-bcf8b188724d/artifacts/019a073d-7fbe-45c5-ae3b-a7e3c7b1a708) Jest Tests #<span></span>3 / EntityAnalyticsManagementPage toggles the save bar when making changes to the closed alerts toggle

### History
* :broken_heart: [Build #351470](https://buildkite.com/elastic/kibana-pull-request/builds/351470) failed b40711d112a874f4b631b61c8ce475de54a85a87
* :broken_heart: [Build #351456](https://buildkite.com/elastic/kibana-pull-request/builds/351456) failed 01ad9fdad8565ac7058db7e3ade5a1752b8956d4
* :broken_heart: [Build #351431](https://buildkite.com/elastic/kibana-pull-request/builds/351431) failed 2cfdcbd369972568e7721a059e98e91a999cd7b2
* :broken_heart: [Build #350580](https://buildkite.com/elastic/kibana-pull-request/builds/350580) failed 9014a177b2527dc62546b9675c1fc9224d0fe7eb
* :broken_heart: [Build #350176](https://buildkite.com/elastic/kibana-pull-request/builds/350176) failed 0243165936395b1324b268a9999f53a873f436f9

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353163","number":353163,"commit":"cb679380cd783a791099e0821f2df1cf32e31ac9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351470","number":351470,"commit":"b40711d112a874f4b631b61c8ce475de54a85a87"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351456","number":351456,"commit":"01ad9fdad8565ac7058db7e3ade5a1752b8956d4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351431","number":351431,"commit":"2cfdcbd369972568e7721a059e98e91a999cd7b2"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350580","number":350580,"commit":"9014a177b2527dc62546b9675c1fc9224d0fe7eb"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/350176","number":350176,"commit":"0243165936395b1324b268a9999f53a873f436f9"}],"number":353163}
buildkite-pr-comment-->

tiansivive committed:
wip

tiansivive committed:
addressing PR feedback

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/jest_integration -u src/core/server/integration_tests/ci_checks'

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-frontend

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/354938)
* Commit: 509477a6f3df4f79468ea547d063a2c469b5e749

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/354938#019a248b-2044-4fa2-9600-acf813d215bc)
* [Check changes in Saved Objects](https://buildkite.com/elastic/kibana-pull-request/builds/354938#019a248b-20a1-4da5-9bdc-de113e1ec793)

### History
* :broken_heart: [Build #353607](https://buildkite.com/elastic/kibana-pull-request/builds/353607) failed 40e7e51b15dfae2d15273c3686a6b70609d2f09c
* :broken_heart: [Build #353476](https://buildkite.com/elastic/kibana-pull-request/builds/353476) failed e21f3808b35f61a16da514ea30f482f5554c8667
* :broken_heart: [Build #353163](https://buildkite.com/elastic/kibana-pull-request/builds/353163) failed cb679380cd783a791099e0821f2df1cf32e31ac9
* :broken_heart: [Build #351470](https://buildkite.com/elastic/kibana-pull-request/builds/351470) failed b40711d112a874f4b631b61c8ce475de54a85a87
* :broken_heart: [Build #351456](https://buildkite.com/elastic/kibana-pull-request/builds/351456) failed 01ad9fdad8565ac7058db7e3ade5a1752b8956d4

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354938","number":354938,"commit":"509477a6f3df4f79468ea547d063a2c469b5e749"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353607","number":353607,"commit":"40e7e51b15dfae2d15273c3686a6b70609d2f09c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353476","number":353476,"commit":"e21f3808b35f61a16da514ea30f482f5554c8667"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353163","number":353163,"commit":"cb679380cd783a791099e0821f2df1cf32e31ac9"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/351470","number":351470,"commit":"b40711d112a874f4b631b61c8ce475de54a85a87"}],"number":354938}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Resolve merge conflicts: Use centralized types from common.ts

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/eslint_all_files --no-cache --fix'

abhishekbhatia1710 committed:
Splitting the hooks file into smaller components and adding relevant unit tests

abhishekbhatia1710 committed:
Merge

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/355583)
* Commit: c600a2a6a0615a9e358f34f61f45096dd3e2a632

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/355583#019a29aa-f9f3-4e34-b440-1ae7d44d46c0)

### History
* :broken_heart: [Build #355038](https://buildkite.com/elastic/kibana-pull-request/builds/355038) failed 88e53c3987e5d18b2ef0311f7e2d965b8b2dd0e4
* :broken_heart: [Build #354938](https://buildkite.com/elastic/kibana-pull-request/builds/354938) failed 509477a6f3df4f79468ea547d063a2c469b5e749
* :broken_heart: [Build #353607](https://buildkite.com/elastic/kibana-pull-request/builds/353607) failed 40e7e51b15dfae2d15273c3686a6b70609d2f09c
* :broken_heart: [Build #353476](https://buildkite.com/elastic/kibana-pull-request/builds/353476) failed e21f3808b35f61a16da514ea30f482f5554c8667
* :broken_heart: [Build #353163](https://buildkite.com/elastic/kibana-pull-request/builds/353163) failed cb679380cd783a791099e0821f2df1cf32e31ac9

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355583","number":355583,"commit":"c600a2a6a0615a9e358f34f61f45096dd3e2a632"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355038","number":355038,"commit":"88e53c3987e5d18b2ef0311f7e2d965b8b2dd0e4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/354938","number":354938,"commit":"509477a6f3df4f79468ea547d063a2c469b5e749"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353607","number":353607,"commit":"40e7e51b15dfae2d15273c3686a6b70609d2f09c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353476","number":353476,"commit":"e21f3808b35f61a16da514ea30f482f5554c8667"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/353163","number":353163,"commit":"cb679380cd783a791099e0821f2df1cf32e31ac9"}],"number":355583}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
CHanges to the import path

abhishekbhatia1710 committed:
Resolving merge conflicts

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-frontend

abhishekbhatia1710 committed:
Changes as per new figma designs and discussions

abhishekbhatia1710 committed:
Fixing registered types

abhishekbhatia1710 commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/alert_filters_kql_bar.tsx:
For now, i am keeping the KQL strings as is, as discussed with Tiago

abhishekbhatia1710 committed:
Adding unit tests for alert_filters_kql_bar

kibanamachine committed:
Changes from node scripts/eslint_all_files --no-cache --fix

abhishekbhatia1710 committed:
Fixing tests

abhishekbhatia1710 committed:
Merge branch 'ea-13606-alert-filtering-risk-score-frontend' of github.com:abhishekbhatia1710/kibana into ea-13606-alert-filtering-risk-score-frontend

abhishekbhatia1710 committed:
Fixing tests

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-frontend

abhishekbhatia1710 committed:
Fixing more tests

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356552)
* Commit: 71727cb82afae928d331c53bceb5f97501f99057

### Failed CI Steps
* [FTR Configs #21](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-87f7-4695-a34e-0efedb8ae3af)
* [FTR Configs #32](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8805-490b-b8a2-755df95758f5)
* [FTR Configs #33](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8806-4132-9c2e-5939040784c4)
* [FTR Configs #56](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8823-471d-b95c-aaddb29b9233)
* [FTR Configs #57](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8824-4031-a19a-74e3ec29df16)
* [FTR Configs #64](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8830-490e-8b46-cb7597c5ab5b)
* [FTR Configs #72](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-883a-4f17-bc70-56784990db0b)
* [FTR Configs #78](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8842-49c0-9a5b-7911274e5ac7)
* [FTR Configs #94](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-885b-4c7c-9c23-f91e287284a4)
* [FTR Configs #99](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-8862-493b-8647-8c3ce546ecdb)
* [FTR Configs #105](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-886a-4c2f-ba21-27500bb178a7)
* [FTR Configs #108](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-886e-4cf4-b44c-535adc5171c2)
* [FTR Configs #121](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-887e-44d0-8f94-d5638359c660)
* [FTR Configs #122](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-887f-4ef4-9034-d67b8ccdade1)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356552#019a3458-87f7-4695-a34e-0efedb8ae3af) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356552/jobs/019a3458-87f7-4695-a34e-0efedb8ae3af/artifacts/019a3488-0d35-4f9d-a8c3-384b6e9fe064) FTR Configs #<span></span>21 / serverless observability UI - Cases and Rules Serverless Observability Cases Case View filter activity filters by history successfully

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7b68136a62fd4625d1b3baa9bfe09fde359a6895) | [after](https://github.com/elastic/kibana/commit/71727cb82afae928d331c53bceb5f97501f99057) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8315 | 8320 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7b68136a62fd4625d1b3baa9bfe09fde359a6895) | [after](https://github.com/elastic/kibana/commit/71727cb82afae928d331c53bceb5f97501f99057) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 11.0MB | 11.0MB | +8.9KB |



### History
* :broken_heart: [Build #356534](https://buildkite.com/elastic/kibana-pull-request/builds/356534) failed df6f40c570dce8acf9b964f3ba6d40d8f7afe20a
* :broken_heart: [Build #356347](https://buildkite.com/elastic/kibana-pull-request/builds/356347) failed c98bfc0e74df6ff4a6e88f21bddfb33e5f1f89da
* :broken_heart: [Build #356250](https://buildkite.com/elastic/kibana-pull-request/builds/356250) failed fc1d9bae54a40bea6850a87dd9c3c4f83cf71a32
* :broken_heart: [Build #356160](https://buildkite.com/elastic/kibana-pull-request/builds/356160) failed 8cea137a905478d78de390a492f73c54cc079b04
* :broken_heart: [Build #355583](https://buildkite.com/elastic/kibana-pull-request/builds/355583) failed c600a2a6a0615a9e358f34f61f45096dd3e2a632

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356552","number":356552,"commit":"71727cb82afae928d331c53bceb5f97501f99057"},{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356534","number":356534,"commit":"df6f40c570dce8acf9b964f3ba6d40d8f7afe20a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356347","number":356347,"commit":"c98bfc0e74df6ff4a6e88f21bddfb33e5f1f89da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356250","number":356250,"commit":"fc1d9bae54a40bea6850a87dd9c3c4f83cf71a32"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356160","number":356160,"commit":"8cea137a905478d78de390a492f73c54cc079b04"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/355583","number":355583,"commit":"c600a2a6a0615a9e358f34f61f45096dd3e2a632"}],"number":356552}
buildkite-pr-comment-->

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-frontend

machadoum commented on x-pack/solutions/security/plugins/security_solution/public/entity_analytics/components/risk_score_management/alert_filters_kql_bar.tsx:
I noticed this code was copied and pasted from other KQlBar implementations. So I am keeping the comment resolved.

machadoum reviewed (approved):
Great job! 🚀 🚀 🚀

abhishekbhatia1710 committed:
Addressing review comments

- The hook now imports the generated server types instead of redefining them
- cast in the preview hook is removed

abhishekbhatia1710 committed:
Merge branch 'ea-13606-alert-filtering-risk-score-frontend' of github.com:abhishekbhatia1710/kibana into ea-13606-alert-filtering-risk-score-frontend

abhishekbhatia1710 commented:
Thanks @machadoum , i have addressed the review comments in the recent commit. 

- The hook now imports the generated server types instead of redefining them
- cast in the preview hook is removed

abhishekbhatia1710 committed:
Merge branch 'main' into ea-13606-alert-filtering-risk-score-frontend

elasticmachine commented:
## :yellow_heart: Build succeeded, but was flaky
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/356805)
* Commit: 5d2b7d8e5cbfe632c92e8091c445bfdc0d17d36f

### Failed CI Steps
* [Scout Test Run Builder](https://buildkite.com/elastic/kibana-pull-request/builds/356805#019a35ab-5dec-4e8d-91bd-16a8aff98ab3)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/356805#019a35ab-5dec-4e8d-91bd-16a8aff98ab3) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/356805/jobs/019a35ab-5dec-4e8d-91bd-16a8aff98ab3/artifacts/019a35cc-3456-453a-b1e3-cc5f115fdd2b) Scout Test Run Builder / serverless-security - EUI testing wrapper: EuiDataGrid - data grid, run

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/0359ff6359a58a9f257302f63b763fdeca980ee8) | [after](https://github.com/elastic/kibana/commit/5d2b7d8e5cbfe632c92e8091c445bfdc0d17d36f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 8315 | 8320 | +5 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/0359ff6359a58a9f257302f63b763fdeca980ee8) | [after](https://github.com/elastic/kibana/commit/5d2b7d8e5cbfe632c92e8091c445bfdc0d17d36f) | diff |
| --- | --- | --- | --- |
| `securitySolution` | 11.0MB | 11.0MB | +9.4KB |



### History
* :broken_heart: [Build #356633](https://buildkite.com/elastic/kibana-pull-request/builds/356633) failed 3b70e0432127c2269f8b0c75d330892456b15f85
* :yellow_heart: [Build #356552](https://buildkite.com/elastic/kibana-pull-request/builds/356552) was flaky 71727cb82afae928d331c53bceb5f97501f99057
* :broken_heart: [Build #356534](https://buildkite.com/elastic/kibana-pull-request/builds/356534) failed df6f40c570dce8acf9b964f3ba6d40d8f7afe20a
* :broken_heart: [Build #356347](https://buildkite.com/elastic/kibana-pull-request/builds/356347) failed c98bfc0e74df6ff4a6e88f21bddfb33e5f1f89da
* :broken_heart: [Build #356250](https://buildkite.com/elastic/kibana-pull-request/builds/356250) failed fc1d9bae54a40bea6850a87dd9c3c4f83cf71a32

cc @tiansivive @abhishekbhatia1710

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356805","number":356805,"commit":"5d2b7d8e5cbfe632c92e8091c445bfdc0d17d36f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356633","number":356633,"commit":"3b70e0432127c2269f8b0c75d330892456b15f85"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356552","number":356552,"commit":"71727cb82afae928d331c53bceb5f97501f99057"},{"buildStatus":{"state":"running","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356534","number":356534,"commit":"df6f40c570dce8acf9b964f3ba6d40d8f7afe20a"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356347","number":356347,"commit":"c98bfc0e74df6ff4a6e88f21bddfb33e5f1f89da"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/356250","number":356250,"commit":"fc1d9bae54a40bea6850a87dd9c3c4f83cf71a32"}],"number":356805}
buildkite-pr-comment-->
